### ISSUE TITLE
libuv assert failure

### ISSUE THREAD

**Issue by PatriceJiang** [warmcat/libwebsockets#1390]
Hi all. 

libuv: 1.23.1
libwebsockets: 2.4.2
test platform: win10

- libuv is enabled.  
- connecting to `ws://invalid.url.com/` (which will redirect to `https://`)

lead to error: 
```
Assertion failed: handle->flags & UV_HANDLE_CLOSING, file e:\github\libuv\src\win\poll.c, line 636
```

it is fine when connected to `wss://invalid.url.com/`

here is the test code on win32 https://github.com/PatriceJiang/libwebsocket_libuv_demo/blob/master/Main.cpp

**Comment by PatriceJiang** [comment]
On the master, symbol `lws_uv_initloop` is renamed to `elops_init_pt_uv` and it's not exported. 
will libuv be removed from `libwebsockets`?

**Comment by PatriceJiang** [comment]
I have just update my code, and a similiar occurs. 
```
Assertion failed: !(handle->flags & UV_HANDLE_CLOSING), file e:\github\libuv\src\win\poll.c, line 231
```

**Comment by lws-team** [comment]
Okay... it sounds like it is unrelated to windows platform... basically it is complaining that libuv is being asked to close something twice... does thing bring anything to mind about what you have to do to reproduce it, or your code?

If you build the minimal example I pointed to before, doesn't it have the same problem?

**Comment by PatriceJiang** [comment]
What I did in my code is connecting to a `non-websocket` URL (ws://invalid.url.com), which return 301 and redirect to  https://invalid.url.com, it's still `non-websocket`.

**Comment by lws-team** [comment]
I see... I'll try it later thismorning.

**Comment by lws-team** [comment]
It's a lot easier for me to look at this stuff with a) a backtrace, and b) a reproducer.

I adapted the ws-client minimal example to be able to work with libuv with a flag, and to be able to point to a given address (and be ready for https redirect).  Then I was able to reproduce it by pointing it to :80 on invalid.url.com, it was redirected to https on :443 and blew while the wsi was being "reset".

Basically libuv's two-stage async close meant the close was still trying to use the watcher in the wsi that was "reset" to target the redirect address.  It's solved by having the watcher duplicated into an allocation in the libuv role "manual close" and freeing it in the libuv role "manual close callback".  Valgrind remains happy after a few retries so I think this is solved.

Patches on master and v3.0-stable.
